luma/core/device.py

```diff
@@ -90,6 +91,61 @@ def cleanup(self):
         self._serial_interface.cleanup()
 
 
+class parallel_device(device):
+    """
+    Wrapper class to manage communications with devices that can operate in
+    four or eight bit modes.
+
+    .. versionadded:: 1.15.1
+    """
+
+    def __init__(self, const=None, serial_interface=None, exec_time=1e-6 * 50, **kwargs):
+        super(parallel_device, self).__init__(const, serial_interface)
+
+        self._exec_time = exec_time
+        self._bitmode = serial_interface._bitmode if hasattr(serial_interface, '_bitmode') else 4
+        assert self._bitmode in (4, 8), 'Bit mode {0} is invalid.  It can only be 4 or 8'.format(self._bitmode)
+
+    def _eight_to_four(self, data):
+        """
+        Convert list of bytes (8 bit values) into a list of nibbles (4 bit values)
+        """
+        return [f(x) for x in data for f in (lambda x: x >> 4, lambda x: 0x0F & x)]
+
+    def command(self, *cmd, exec_time=None, only_low_bits=False):
+        """
+        Sends a command or sequence of commands through to the serial interface.
+        If operating in four bit mode, expands each command from one byte
+        values (8 bits) to two nibble values (4 bits each)
+
```

Comment:
I think the docstring should mention `sleep` is being used here (aka blocking).

Reply:
Will add the warning.   The delays are typically small but if they become a problem we could move to a separate thread or multi-process.  Should the same notice show up in the driver doc strings?

Reply:
the more descriptive the better imo. and it'll feel even better when eventually in the future, some async solution will allow us to delete these words from the docstring ;)